src/main/java/com/algorand/algosdk/templates/ContractTemplate.java

```diff
@@ -166,4 +166,18 @@ protected static ContractTemplate inject(byte [] program, List<ParameterValue> v
 
 		return new ContractTemplate(updatedProgramByteArray);
 	}
+
+	protected static Logic.ProgramData readAndVerifyContract(byte[] program, int numInts, int numByteArrays) {
+		try {
+			Logic.ProgramData data = Logic.readProgram(program, null);
+
+			if (!data.good || data.intBlock.size() != numInts || data.byteBlock.size() != numByteArrays) {
+				throw new IllegalArgumentException("Invalid contract detected.");
+			}
+
+			return data;
+		} catch (Exception e) {
+			throw new IllegalArgumentException("Unable to parse contract.");
```

Comment:
Is ignoring the thrown exception and throwing a new one the desired behavior, as opposed to letting the thrown exception bubble up? ("Yes" is a perfectly ok answer :+1:)

Reply:
Oops, no I wanted the different messages.